Zend/tests/arrow_functions/005.phpt

```diff
@@ -10,25 +10,47 @@ class Test {
         $r = new ReflectionFunction($fn);
         var_dump($r->getClosureThis());
 
+        $fn = fn() => {};
+        $r = new ReflectionFunction($fn);
+        var_dump($r->getClosureThis());
+
         $fn = fn() => $this;
         var_dump($fn());
 
+        $fn = fn() => { return $this; };
+        var_dump($fn());
+
         $fn = fn() => Test::method2();
         $fn();
 
+        $fn = fn() => { return Test::method2(); };
+        $fn();
+
         $fn = fn() => call_user_func('Test::method2');
         $fn();
 
+        $fn = fn() => { return call_user_func('Test::method2'); };
+        $fn();
+
         $thisName = "this";
         $fn = fn() => $$thisName;
         var_dump($fn());
 
+        $fn = fn() => { return $$thisName; };
+        var_dump($fn());
+
         $fn = fn() => self::class;
         var_dump($fn());
 
+        $fn = fn() => { return self::class; };
+        var_dump($fn());
+
         // static can be used to unbind $this
         $fn = static fn() => isset($this);
         var_dump($fn());
+
+        $fn = static fn() => { return isset($this); };
+        var_dump($fn());
```

Comment:
add one more static test and check with reflection that `$this` is really not bound

Reply:
@mvorisek Gonna wait for the RFC decision before updating that.